kube/services/pidgin/pidgin-deploy.yaml

```diff
@@ -0,0 +1,90 @@
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: pidgin-deployment
+spec:
+  selector:
+    # Only select pods based on the 'app' label
+    matchLabels:
+      app: pidgin
+  revisionHistoryLimit: 2
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 1
+      maxUnavailable: 0
+  template:
+    metadata:
+      labels:
+        app: pidgin
+        GEN3_DATE_LABEL
+    spec:
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+          - weight: 100
+            podAffinityTerm:
+              labelSelector:
+                matchExpressions:
+                - key: app
+                  operator: In
+                  values:
+                  - pidgin
+              topologyKey: "kubernetes.io/hostname"
+      automountServiceAccountToken: false
+      volumes:
+        - name: config-volume
+          secret:
+            secretName: "pidgin-secret"
+        - name: config-helper
+          configMap:
+            name: config-helper
+        - name: cert-volume
+          secret:
+            secretName: "cert-pidgin-service"
+        - name: ca-volume
+          secret:
+            secretName: "service-ca"
+      containers:
+        - name: pidgin
+          GEN3_PIDGIN_IMAGE
+          livenessProbe:
+            httpGet:
+              path: /_status
+              port: 80
+            initialDelaySeconds: 30
+            periodSeconds: 60
+            timeoutSeconds: 30
+          readinessProbe:
+            httpGet:
+              path: /_status
+              port: 80
+          ports:
+          - containerPort: 80
+          - containerPort: 443
+          volumeMounts:
+            - name: "config-volume"
+              readOnly: true
+              mountPath: "/var/www/pidgin/wsgi.py"
+              subPath: "wsgi.py"
+            - name: "config-helper"
+              readOnly: true
+              mountPath: "/var/www/pidgin/config_helper.py"
+              subPath: config_helper.py
+            - name: "cert-volume"
+              readOnly: true
+              mountPath: "/mnt/ssl/service.crt"
+              subPath: "service.crt"
+            - name: "cert-volume"
+              readOnly: true
+              mountPath: "/mnt/ssl/service.key"
+              subPath: "service.key"
+            - name: "ca-volume"
+              readOnly: true
+              mountPath: "/mnt/ssl/cdis-ca.crt"
+              subPath: "ca.pem"
+          imagePullPolicy: Always
+          resources:
+            limits:
+              cpu: 0.8
+              memory: 128Mi
```

Comment:
Does this service use `authutils` to download fence public keys?  If so, then you need to register the certificate authority with the runtime for the code to work with self signed certificates and internal cert authorities ...